### PR TITLE
engine_test: use simpler stdlib to copy dir

### PR DESCRIPTION
**Pull request by dt**
We don't need to pull in a dep just to copy a dir in a test.

The upstream code does handle more edge cases (perms, symlinks on both sides, etc),
but for this simple test we don't actually need the most general solution.

### CODE REVIEW

**Comment by tamird** [comment]
How come `vendor` didn't change? I see it has a new ref, but https://github.com/cockroachdb/vendored/compare/9ec36a627255186815164ee025b4fa8f1c2a98ea...c91d3cae26eaee7737ff139ae92aed08449d2561 comes up empty (and in reverse, too).

File: pkg/storage/engine/bench_test.go [hunk 1]
```diff
@@ -512,6 +512,29 @@ func BenchmarkMVCCMergeTimeSeries_RocksDB(b *testing.B) {
 	runMVCCMerge(setupMVCCInMemRocksDB, &value, 1024, b)
 }
 
+func copyDir(from, to string) error {
+	return filepath.Walk(from, func(srcPath string, info os.FileInfo, err error) error {
```
Comment (tamird) [inline pkg/storage/engine/bench_test.go]: what is the "err" argument for? should it be handled?

**Review by tamird — COMMENTED** [review]
Can you summarize what this dep does (which has considerably more complicated code) that this alternative does not? Unfortunately I can't reason about the pitfalls of this implementation without thoroughly reviewing the vendored code, which makes me nervous about this change not handling edge cases in our test environments.

**Comment by dt** [comment]
Looks like ref wasn't on `vendored` yet -- is now if you refresh.

File: pkg/storage/engine/bench_test.go (hunk 1 above)
Comment (dt) [inline pkg/storage/engine/bench_test.go]: yeah, good call. done.

**Review by tamird — APPROVED** [review]

**Comment by dt** [comment]
I mentioned in the commit message some of the things the dep was doing that we're not, but the the simpler impl is enough for the test (I verified `make bench PKG=./pkg/storage/engine BENCHES=BenchmarkMVCCDeleteRange_RocksDB` passed)

**Comment by tamird** [comment]
Ack, thanks.
